src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+                result.privateItems = await this.getFeedPrivateItems(feedInfoName)
+            }
+
+        } catch (e) {
+            logger.error(e);
+        }
+
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedItems', {feedInfoName});
+        const items = await this.feeds.get(feedInfoName).get('items').then();
+        if (items) {
+            for (let keyItem in items) {
+                if (keyItem !== '_') {
+                    const mainFields = await this.feeds
+                        .get(feedInfoName)
+                        .get('items')
+                        .get(keyItem).map();
+                    result[keyItem] = _.pick(mainFields, this.feedItemFields);
+                    const dataItem = await this.feeds
+                        .get(feedInfoName)
+                        .get('items')
+                        .get(keyItem)
+                        .map()
+                        .get('data')
+                    result[keyItem].data = _.pick(dataItem, this.feedItemDataFields)
+                }
+            }
+        }
+        return result
+    }
+
+    /**
+     * Return array items
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeedPrivateItems(feedInfoName: string): Promise<FeedItem | any> {
+        let result: any = {};
+        logger.info('class getFeedPrivateItems', {feedInfoName});
+        const privateItems = await this.feeds.get(feedInfoName).get('privateItems').then();
+        if (privateItems) {
+            for (let privateKey in privateItems) {
+                if (privateKey !== '_') {
+                    result[privateKey] = {};
+                    let items = await this.feeds
+                        .get(feedInfoName)
+                        .get('privateItems')
+                        .get(privateKey)
+                        .then();
+                    if (items) {
+                        for (let keyItem in items) {
+                            if (keyItem !== '_') {
+                                const mainFields = await this.feeds
+                                    .get(feedInfoName)
+                                    .get('privateItems')
+                                    .get(privateKey)
+                                    .get(keyItem)
+                                    .map()
+                                console.log('**********************')
+                                console.log(mainFields)
+                                console.log('**********************')
+                                result[privateKey][keyItem] = _.pick(mainFields, this.feedItemFields);
+                                const dataItem = await this.feeds
+                                    .get(feedInfoName)
```

Comment:
there is a .path() shortener allows to avoid log .get() chains
https://gun.eco/docs/API#-a-name-path-a-gun-path-key-